packages/lib/resultShareUrl/service.ts

```diff
@@ -0,0 +1,126 @@
+import { Prisma } from "@prisma/client";
+import { customAlphabet } from "nanoid";
+
+import { prisma } from "@formbricks/database";
+import { ZId } from "@formbricks/types/environment";
+import { DatabaseError, ResourceNotFoundError } from "@formbricks/types/errors";
+
+import { validateInputs } from "../utils/validate";
+
+// Create the short url and return it
+export const createResultShareKey = async (surveyId: string): Promise<string> => {
+  validateInputs([surveyId, ZId]);
+
+  try {
+    const survey = await prisma.survey.findFirst({
+      where: {
+        id: surveyId,
+      },
+    });
+
+    if (!survey) {
+      throw new ResourceNotFoundError("survey", surveyId);
+    }
+
+    if (survey.resultShareKey) {
+      return survey.resultShareKey;
+    }
+
+    const id = customAlphabet("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789", 10)();
```

Comment:
please increase length to 20 to increase the security since here it's easier to brute force.